scylla/src/transport/connection_pool.rs

```diff
@@ -80,6 +90,43 @@ enum PoolConnections {
     },
 }
 
+fn format_connection_vector(conns: &[Arc<Connection>]) -> String {
```

Comment:
Could you try converting this and `format_sharded_connection_vector` to a version which uses a `std::fmt::Formatter`, just like the impls for `std::fmt::Display`? It will be more efficient this way, and probably not much more complicated.

While I am aware that this code is for the debugging purposes, I think we should not make them unnecessarily inefficient. Those functions will probably be called frequently if trace level is enabled, so let's make sure that they don't add too much overhead. This is especially important if somebody wants to use it to debug production issues ;)

Reply:
yup, sure, I'll rewrite to std::fmt::Formatter; It's a pity I can't just implement Display for `Vec<Arc<MyType>>`

Reply:
If you want to use `Display` then you can introduce a newtype wrapper:

```rust
enum ConnectionVectorWrapper<'a>(&'a Vec<Arc<Connection>>);

impl<'a> std::fmt::Display for ConnectionVectorWrapper<'a> {
    // ...
}

// ...

trace!("My connections: {}", ConnectionVectorWrapper(&conns));
```